change recommendation to stash changes (#113)
Ada-Developers-Academy/core-unit-2: api-1-setup-read/intro-to-hello-books.md

1. Rename this project folder to `ada-hello-books-api` to make it clear it's the Ada maintained version.
1. When beginning a lesson, checkout the corresponding starting branch for that lesson.
1. Code along with the lesson **as much or as little as best supports your learning**. You may find it's best to watch all the videos without coding along. Alternatively, you may prefer to implement every new feature along with the reading and videos.

> 1. When finishing a lesson, 
>    - you may `add` and `commit` your changes before checking out the next branch
>    - alternatively, you may remove any changes you've made using `git stash`
>       - `git stash` resets files to the previous commit point. 

Reviewer: Some thoughts on some possible content here. I think it's important to add the `-u` to stash, since stash doesn't deal with untracked by default, which the `app/__init__.py` will be, and which conflicts with the next branch they need to view.

I'd harden the language to definitely do one or the other. Currently it sounds like they could choose to do neither, which would cause a conflict.

I feel like it would be good to give students a reason to pick one over the other. With as brief as this is, I don't think they have a clear enough reason to prefer one or the other. Because this is in bullets, it's hard to fit in extra explanation, so maybe another callout?

```suggestion
1. After finishing a lesson, you will not be able to switch branches until Git knows what to do with your existing changes. You should either:
   - `add` and `commit` your changes, or
   - use `git stash -u` to set aside any changes you've made
   
### !callout-info

## When to Commit, When to Stash

Committing your changes will add them to the local commit history. If you switch to a different branch, and then return to the branch in which you committed, you will see your changes as you left them. If you want to continue to see your changes in the branch, you should add and commit your changes.

<br>

Stashing your changes will store them in a temporary commit area, and the contents of the branch will revert back to what they looked like before you started working. If you switch to a different branch, and then return to the branch in which you were working, it will appear as though you did no work in that branch. If you want the branch to return to its starting point, you should stash your changes.

### !end-callout   

```

Author: Looks good! Do you think we should mention anything about `git stash pop`

Reviewer: Maybe like "If you'd like to restore changes that have been stashed, git provides several ways to do this. A good starting point is `git stash pop`. Follow your curiosity!"

Gotta have a FYC in there! 😆

Author: I'm going to hold off on merging this in until we have more perspective from how the Building an API lessons go!

1. Rename this project folder to `ada-hello-books-api` to make it clear it's the Ada maintained version.
1. When beginning a lesson, checkout the corresponding starting branch for that lesson.
1. Code along with the lesson **as much or as little as best supports your learning**. You may find it's best to watch all the videos without coding along. Alternatively, you may prefer to implement every new feature along with the reading and videos.
1. After finishing a lesson, you will not be able to switch branches until Git knows what to do with your existing changes. You should either:

>    - `add` and `commit` your changes, or
>    - use `git stash -u` to set aside any changes you've made

Author: ```suggestion
   - `add` and `commit` your changes, or
   - use `git stash -u` to set aside any changes you've made

If you'd like to restore changes that have been stashed, Git provides several ways to do this. A good starting point for investigation is git stash pop. Follow your curiosity!
```

1. Rename this project folder to `ada-hello-books-api` to make it clear it's the Ada maintained version.
1. When beginning a lesson, checkout the corresponding starting branch for that lesson.
1. Code along with the lesson **as much or as little as best supports your learning**. You may find it's best to watch all the videos without coding along. Alternatively, you may prefer to implement every new feature along with the reading and videos.
1. After finishing a lesson, you will not be able to switch branches until Git knows what to do with your existing changes. You should either:
   - `add` and `commit` your changes, or
   - use `git stash -u` to set aside any changes you've made

> If you'd like to restore changes that have been stashed, Git provides several ways to do this. A good starting point for investigation is git stash pop. Follow your curiosity!

Reviewer 2: I think it could be helpful to surround "git stash pop" in quotes to stand out from the other text, perhaps putting it in code ticks as well.

```suggestion
If you'd like to restore changes that have been stashed, Git provides several ways to do this. A good starting point for investigation is `"git stash pop"`. Follow your curiosity!
```

Author: I think just code ticks make the most sense. Thanks!
